File: maximum-subarray/maximum_subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    result = sum = nums[0];
    for (var i = 1; i < nums.length; i ++) {
        if (sum < 0) {
            sum = nums[i];
        } else {
            sum += nums[i];
        }
        if (result < sum) {
            result = sum;
        }
    }
    return result;
};

File: maximum-subarray/maximum_subarray.go
func maxSubArray(nums []int) int {
    sum := nums[0]
    result := nums[0]
    for k, v := range nums {
        if k == 0 {
            continue
        }
        if sum < 0 {
            sum = v
        } else {
            sum += v
        }
        if result < sum {
            result = sum
        }
    }
    return result
}